routing/car_router.cpp

```diff
@@ -599,7 +599,8 @@ IRouter::ResultCode CarRouter::FindSingleRouteDispatcher(FeatureGraphNode const
     }
     LOG(LINFO, (m_router->GetName(), "route from", MercatorBounds::ToLatLon(source.segmentPoint),
                 "to", MercatorBounds::ToLatLon(target.segmentPoint)));
-    result = m_router->CalculateRoute(source.mwmId, source.segmentPoint,
+    m_router->SetCountry(source.mwmId.GetInfo()->GetCountryName());
+    result = m_router->CalculateRoute(source.segmentPoint,
```

Comment:
What's the point to split CalculateRoute into two calls - to SetCountry() and to CalculateRoute()? Why not to keep old notation?